solutions/45.js

```diff
@@ -0,0 +1,15 @@
+const kebab = (a,i=0,b=[])=>{
+  if(i == a.length){
+    b=b.join('');
+    return b;
+  }
+  if(a[i]!=a[i].toUpperCase()){
+    b.push(a[i]);
+  }else{
+    c=a[i].toLowerCase();
+    b.push("-");
+    b.push(c);
```

Comment:
You are pushing a.toLowercase no matter what, you push additional '-' when it's uppercase. This way you don't need else statement at all.